modin/experimental/core/execution/native/implementations/omnisci_on_native/interchange/dataframe_protocol/column.py

```diff
@@ -194,12 +195,17 @@ def describe_categorical(self) -> Dict[str, Any]:
             col = col.combine_chunks()
 
         col = col.chunks[0]
-        mapping = dict(enumerate(col.dictionary.tolist()))
+        cat_frame = Series(col.dictionary.tolist())._query_compiler._modin_frame
```

Comment:
```suggestion
        cat_frame = OmnisciOnNativeDataframe.from_arrow(
            pyarrow.Table.from_pydict({self._col.column_names[0]: col.dictionary})
        )
```
Can do a zero-copy here which is significantly faster with an increasing amount of categories
```
======= ncategories=1000 =======
        Using low-level `.from_arrow()`: 0.00043630000004668545
        Using high-level `Series()`: 0.0025355000000217842
======= ncategories=10000 =======
        Using low-level `.from_arrow()`: 0.0003973999998834188
        Using high-level `Series()`: 0.007733099999995829
======= ncategories=1000000 =======
        Using low-level `.from_arrow()`: 0.0008338000000094326
        Using high-level `Series()`: 0.6584339000000909
```

<details><summary>repro</summary>

```python
import modin.config as cfg

cfg.StorageFormat.put("Omnisci")

from timeit import default_timer as timer
import modin.pandas as pd
import pyarrow as pa
import pandas

# On my machine this import causes segfault for some reason...
# from modin.experimental.core.execution.native.implementations.omnisci_on_native.dataframe.dataframe import OmnisciOnNativeDataframe
# So using this weird way to get the class...
OmnisciOnNativeDataframe = type(pd.DataFrame([])._query_compiler._modin_frame)

def compare(ncategories):
    pa_at = pa.Table.from_pandas(
        pandas.DataFrame({"col": pandas.Categorical([f"val{i}" for i in range(ncategories)])})
    )
    pa_col = pa_at.column(0).chunks[0]

    print(f"======= {ncategories=} =======")

    t1 = timer()
    frame1 = OmnisciOnNativeDataframe.from_arrow(
        pa.Table.from_pydict({"__reduced__": pa_col.dictionary})
    )
    print(f"\tUsing low-level `.from_arrow()`: {timer() - t1}")

    t1 = timer()
    frame2 = pd.Series(pa_col.dictionary.tolist())._query_compiler._modin_frame
    print(f"\tUsing high-level `Series()`: {timer() - t1}")


compare(ncategories=1_000)
compare(ncategories=10_000)
compare(ncategories=1_000_000)
```

</details>

Though, not pushing to apply this in the current PR as the PR is more about compatibility when the suggestion is for optimization.